File: UI Automation Java/Exercise 3 - Automation Java/automation_3/src/main/java/automation_TestNG/BrowserInteraction.java
package automation_TestNG;

import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class BrowserInteraction {
	WebDriver driver;
	public WebElement findElement(WebDriver driver, By xpath) {
		new WebDriverWait(driver,1).until(ExpectedConditions.presenceOfElementLocated(xpath));
		return driver.findElement(xpath);
	}

	public void SendText(WebElement element, String value) {
		element.sendKeys(value);		
	}

	public void Click(WebElement element) {
		element.click();
	}
	
	public void displayed(WebElement element){
		Assert.assertTrue(element.isDisplayed());
		System.out.println("ON SCREEN!!");
	}

	public void isEquals(String str1, String str2){
		Assert.assertEquals(str1,str2);
		System.out.println("PASSED!!");
	}
	
	public void get(String url){
		try{
			driver.get(url);
		}catch(Exception e){
			e.printStackTrace();
		}
	}
	
	public void Close(){
		try{
			driver.close();
		}catch(Exception e){
			e.printStackTrace();
		}
	}

	
}

File: UI Automation C#/Exercise 3 Automation C#/Exercise 3 MS Test/Exercise 3 MS Test/BrowserInteraction.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SeleniumExtras.WaitHelpers;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Exercise_3_MS_Test
{
    class BrowserInteraction
    {     
        public IWebElement findElement(IWebDriver driver, By xpath)
        {           
            return new WebDriverWait(driver, new TimeSpan(0, 0, 3)).Until(x => x.FindElement(xpath));
        }
        public void Click(IWebElement element)
        {
            element.Click();
        }

        public void SendText(IWebElement element, string value)
        {
            element.SendKeys(value);
        }
        

        public void displayed(IWebElement element)
        {
            Assert.IsTrue(element.Displayed);
            Console.WriteLine("ON SCREEN!!");
        }

        
        public void isEquals(String str1, String str2)
        {
            Assert.AreEqual(str1, str2);
            Console.WriteLine("PASSED!!");
        }        
    }
}
